backend/algorithms/objects/conditions.py

```diff
@@ -109,20 +111,29 @@ def set_logic(self, logic: Logic):
 
     def validate(self, user: User) -> tuple[bool, list[str]]:
         """ Returns True if the user is taking these courses in the same term """
+        
         match self.logic:
             case Logic.AND:
-                return all(
+                met = all(
                     user.has_taken_course(course)
                     or user.is_taking_course(course)
                     for course in self.courses
-                ), []
+                )
+                warning = []
+                for course in self.courses:
```

Comment:
this should also be a list comp :)
```
warning = [course for course in self.courses if not user.has_taken_course(course) and not user.is_taking_course(course)]
```

Reply:
or better yet, we can use this to replace `met`, so we dont run the same calculation twice.
if the warnings list is empty, we know its true, else false.